rs/index/src/collection/collection.rs

```diff
@@ -847,31 +860,50 @@ impl<Q: Quantizer + Clone + Send + Sync + 'static> Collection<Q> {
         &self.all_segments
     }
 
-    pub fn remove(&self, user_id: u128, doc_id: u128, sequence_number: u64) -> Result<()> {
-        {
-            let mutable_segments = self.mutable_segments.read();
-            mutable_segments
-                .mutable_segment
-                .read()
-                .invalidate(user_id, doc_id, sequence_number)?;
+    fn remove_impl(
```

Comment:
probably add `#[inline(always)]` here - (I don't trust Rust compiler to automatically do it)